crates/providers/lattice-controller/src/lib.rs

```diff
@@ -0,0 +1,630 @@
+                v.into_iter()
+                    .map(|h| Host {
+                        cluster_issuers: h.cluster_issuers,
+                        ctl_host: h.ctl_host,
+                        id: h.id,
+                        js_domain: h.js_domain,
+                        labels: h.labels,
+                        lattice_prefix: h.lattice_prefix,
+                        prov_rpc_host: h.rpc_host.clone(),
+                        rpc_host: h.rpc_host,
+                        uptime_human: h.uptime_human,
+                        uptime_seconds: h.uptime_seconds,
+                        version: h.version,
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?)
+    }
+
+    /// Retrieve inventory for a given host on the lattice
+    #[instrument(level = "debug", skip(self, _ctx, arg), fields(actor_id = ?_ctx.actor, host = ?arg.host_id, lattice_id = ?arg.lattice_id))]
+    async fn get_host_inventory(
+        &self,
+        _ctx: Context,
+        arg: GetHostInventoryRequest,
+    ) -> ProviderInvocationResult<HostInventory> {
+        Ok(self
+            .get_connections()
+            .await
+            .get_client(&arg.lattice_id.to_string())
+            .await?
+            .get_host_inventory(&arg.host_id)
+            .await
+            .map(|hi| HostInventory {
+                actors: hi
+                    .actors
+                    .into_iter()
+                    .map(|ad| ActorDescription {
+                        id: ad.id,
+                        image_ref: ad.image_ref,
+                        instances: ad
+                            .instances
+                            .into_iter()
+                            .map(|ai| ActorInstance {
+                                annotations: ai.annotations,
+                                instance_id: ai.instance_id,
+                                revision: ai.revision,
+                            })
+                            .collect::<Vec<_>>(),
+                        name: ad.name,
+                    })
+                    .collect::<Vec<_>>(),
+                host_id: hi.host_id,
+                labels: hi.labels,
+                providers: hi
+                    .providers
+                    .into_iter()
+                    .map(|pd| ProviderDescription {
+                        annotations: pd.annotations,
+                        id: pd.id,
+                        image_ref: pd.image_ref,
+                        link_name: pd.link_name,
+                        name: pd.name,
+                        revision: pd.revision,
+                    })
+                    .collect::<Vec<_>>(),
+            })
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?)
+    }
+
+    /// Retrieve claims for a given client
+    #[instrument(level = "debug", skip(self, _ctx), fields(actor_id = ?_ctx.actor, lattice_id))]
```

Comment:
Please `skip_all` here for same reasons mentioned above. Same goes for rest of the file

Reply:
- [ ] Fix `skip_all` for `instrument` call